gluten-core/src/main/scala/io/glutenproject/execution/SortMergeJoinExecTransformer.scala

```diff
@@ -239,10 +225,150 @@ case class SortMergeJoinExecTransformer(
 
   override def getChild: SparkPlan = streamedPlan
 
-  override def doValidate(): Boolean = false
+  def genJoinParametersBuilder(): com.google.protobuf.Any.Builder = {
+    val (isBHJ, isNullAwareAntiJoin, buildHashTableId) = genJoinParameters()
+    // Start with "JoinParameters:"
+    val joinParametersStr = new StringBuffer("JoinParameters:")
+    // isBHJ: 0 for SHJ, 1 for BHJ
+    // isNullAwareAntiJoin: 0 for false, 1 for true
+    // buildHashTableId: the unique id for the hash table of build plan
+    joinParametersStr.append("isBHJ=").append(isBHJ).append("\n")
+      .append("isNullAwareAntiJoin=").append(isNullAwareAntiJoin).append("\n")
+      .append("buildHashTableId=").append(buildHashTableId).append("\n")
+      .append("isExistenceJoin=").append(
+      if (joinType.isInstanceOf[ExistenceJoin]) 1 else 0).append("\n")
+    val message = StringValue
+      .newBuilder()
+      .setValue(joinParametersStr.toString)
+      .build()
+    com.google.protobuf.Any.newBuilder
+      .setValue(message.toByteString)
+      .setTypeUrl("/google.protobuf.StringValue")
+  }
+
+  def genJoinParameters(): (Int, Int, String) = {
+    (-1, -1, "")
+  }
+
+  // Direct output order of substrait join operation
+  protected  val substraitJoinType = joinType match {
+    case Inner =>
+      JoinRel.JoinType.JOIN_TYPE_INNER
+    case FullOuter =>
+      JoinRel.JoinType.JOIN_TYPE_OUTER
+    case LeftOuter | RightOuter =>
+      // The right side is required to be used for building hash table in Substrait plan.
+      // Therefore, for RightOuter Join, the left and right relations are exchanged and the
+      // join type is reverted.
+      JoinRel.JoinType.JOIN_TYPE_LEFT
```

Comment:
I guess this is not true for SMJ because the left table and right table are not exchanged.

Reply:
Removed.